### PR TITLE
Fix MessagePack Integer implementation

### PR DESCRIPTION
**Pull request by SX91**
MessagePack spec declares Int as `[-(2^63-1), 2^64-1]` and encodes negatives (< 0) and non-negatives (>= 0) differently.

`data-msgpack` implementation used Int64 to store both negative and positive values, which in result caused values > 2^63 to be encoded as negative values.

Your tests didn't fail because Word64 -> Int64 -> Word64 would convert just fine. But any other MessagePack implementations would read a negative value (or fail, if positive is expected).

Fixes https://github.com/TokTok/hs-msgpack/issues/15.

### CODE REVIEW

File: src/Data/MessagePack/Generic.hs [hunk 1]
```diff
@@ -37,7 +37,7 @@ instance (GSumPack a, GSumPack b, SumSize a, SumSize b) => GMessagePack (a :+: b
       size = unTagged (sumSize :: Tagged (a :+: b) Word64)
 
   gFromObject = \case
-    ObjectInt code -> checkSumFromObject0 size (fromIntegral code)
+    ObjectWord code -> checkSumFromObject0 size (fromIntegral code)
     o              -> fromObject o >>= uncurry (checkSumFromObject size)
```
Comment (iphydf) [inline src/Data/MessagePack/Generic.hs]: Add space.

File: src/Data/MessagePack/Object.hs [hunk 2]
```diff
@@ -96,6 +101,7 @@ instance Arbitrary Object where
     , ObjectMap    <$> Gen.resize (n `div` 4) arbitrary
     , ObjectExt    <$> arbitrary <*> arbitrary
     ]
+    where negatives = Gen.choose (-0X7FFFFFFFFFFFFFFF, -1)
```
Comment (iphydf) [inline src/Data/MessagePack/Object.hs]: Can you use `minBound` here?

File: test/Data/MessagePackSpec.hs [hunk 3]
```diff
@@ -312,15 +313,19 @@ spec = do
...
   describe "show" $ do
     it "Foo" $ do
-      show (toObject Foo1) `shouldBe` "ObjectInt 0"
-      show (toObject $ Foo3 3) `shouldBe` "ObjectArray [ObjectInt 2,ObjectInt 3]"
-      show (toObject $ Foo9 3 5) `shouldBe` "ObjectArray [ObjectInt 8,ObjectArray [ObjectInt 3,ObjectInt 5]]"
-      show (toObject $ Foo10 3 5 7) `shouldBe` "ObjectArray [ObjectInt 9,ObjectArray [ObjectInt 3,ObjectInt 5,ObjectInt 7]]"
+      show (toObject Foo1) `shouldBe` "ObjectWord 0"
+      show (toObject $ Foo3 3) `shouldBe` "ObjectArray [ObjectWord 2,ObjectWord 3]"
+      show (toObject $ Foo9 3 5) `shouldBe` "ObjectArray [ObjectWord 8,ObjectArray [ObjectWord 3,ObjectWord 5]]"
+      show (toObject $ Foo10 3 5 7) `shouldBe` "ObjectArray [ObjectWord 9,ObjectArray [ObjectWord 3,ObjectWord 5,ObjectWord 7]]"
```
Comment (iphydf) [inline test/Data/MessagePackSpec.hs:327]: Perhaps add or replace one test with a negative number?

**Review by iphydf — CHANGES_REQUESTED** [review]

**Comment by SX91** [comment]
Please review. I've committed all the required changes.

**Review by iphydf — APPROVED** [review]
LGTM, thanks, please squash/rebase.

**Comment by SX91** [comment]
Done.
